vm/src/builtins/builtinfunc.rs

```diff
@@ -140,9 +139,23 @@ impl PyBuiltinFunction {
     }
 }
 
+// PyMethodDescrObject in
+// https://github.com/python/cpython/blob/main/Objects/descrobject.c
 #[pyclass(module = false, name = "method_descriptor")]
 pub struct PyBuiltinMethod {
     value: PyNativeFuncDef,
+    dtype: Option<PyTypeRef>,
```

Comment:
I can make this `PyObjectRef` and store just the type name. 
But, the current implementation is closer to https://github.com/python/cpython/blob/main/Objects/descrobject.c#L43

Reply:
And https://github.com/python/cpython/blob/302cf3550514b174b51b8c24d6e16cb47dc61755/Include/descrobject.h#L43

Reply:
I think its good right now. Staying closer to CPython is generally preferred.

Reply:
I think the corresponding object for `PyBuiltinMethod` in CPython is `PyCMethodObject`.
https://github.com/python/cpython/blob/302cf3550514b174b51b8c24d6e16cb47dc61755/Include/cpython/methodobject.h#L32-L35

And I found a field named `mm_class` with same type. is this related? or is this different field?

Reply:
Yeah, I think that it is more like `PyCMethodObject`, renamed `dtype` to `mmtype` to match the sctructure. 👍